sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -226,13 +252,14 @@ async def serialize_property(value: 'Input[Any]',
 
         if hasattr(value, "assets"):
             asset_archive = cast('AssetArchive', value)
-            obj["assets"] = await serialize_property(asset_archive.assets, deps, input_transformer)
+            obj["assets"] = await serialize_property(asset_archive.assets, deps, input_transformer,
+                keep_output_values=False)
         elif hasattr(value, "path"):
             file_archive = cast('FileArchive', value)
-            obj["path"] = await serialize_property(file_archive.path, deps, input_transformer)
+            obj["path"] = await serialize_property(file_archive.path, deps, input_transformer, keep_output_values=False)
         elif hasattr(value, "uri"):
             remote_archive = cast('RemoteArchive', value)
-            obj["uri"] = await serialize_property(remote_archive.uri, deps, input_transformer)
+            obj["uri"] = await serialize_property(remote_archive.uri, deps, input_transformer, keep_output_values=False)
```

Comment:
False intentional everywhere? Would it matter if it was `keep_output_values` from the param for symmetry?